server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java

```diff
@@ -492,6 +492,13 @@ public class ShuffleServerConf extends RssBaseConf {
                   + HUGE_PARTITION_SIZE_THRESHOLD.key()
                   + "'");
 
+  public static final ConfigOption<Long> MAX_PARTITION_SIZE_THRESHOLD =
+      ConfigOptions.key("rss.server.max-partition.size.threshold")
+          .longType()
+          .defaultValue(Long.MAX_VALUE)
+          .withDescription(
+              "Threshold of max partition size, once exceeding threshold, client will receive an error.");
```

Comment:
This option sets the maximum allowable partition size threshold. If the partition size exceeds this threshold, the client will receive an error message and the transmission of shuffle data will be terminated. This helps to significantly improve the stability of the cluster by preventing partitions from becoming too large.